openmls/src/messages/tests/test_pgs.rs

```diff
@@ -1,98 +1,42 @@
-
-    // Generate KeyPackages
-    let bob_key_package_bundle =
-        KeyPackageBundle::new(&[ciphersuite], &bob_credential_bundle, backend, Vec::new())
-            .expect("An unexpected error occurred.");
-    let bob_key_package = bob_key_package_bundle.key_package();
+    .unwrap();
 
     let alice_key_package_bundle = KeyPackageBundle::new(
         &[ciphersuite],
         &alice_credential_bundle,
         backend,
         Vec::new(),
     )
-    .expect("An unexpected error occurred.");
+    .unwrap();
 
     // Alice creates a group
-    let mut group_alice = CoreGroup::builder(GroupId::random(backend), alice_key_package_bundle)
-        .build(backend)
-        .expect("Could not create group.");
-
-    // Alice adds Bob
-    let bob_add_proposal = group_alice
-        .create_add_proposal(
-            framing_parameters,
-            &alice_credential_bundle,
-            bob_key_package.clone(),
-            backend,
-        )
-        .expect("Could not create proposal.");
-
-    let proposal_store = ProposalStore::from_queued_proposal(
-        QueuedProposal::from_mls_plaintext(ciphersuite, backend, bob_add_proposal)
-            .expect("Could not create QueuedProposal."),
-    );
-    let params = CreateCommitParams::builder()
-        .framing_parameters(framing_parameters)
-        .credential_bundle(&alice_credential_bundle)
-        .proposal_store(&proposal_store)
-        .build();
-    let create_commit_result = match group_alice.create_commit(params, backend) {
-        Ok(c) => c,
-        Err(e) => panic!("Error creating commit: {:?}", e),
+    let group_alice: CoreGroup =
+        CoreGroup::builder(GroupId::random(backend), alice_key_package_bundle)
+            .build(backend)
+            .unwrap();
+
+    let group_info: GroupInfo = group_alice
+        .export_group_info(backend, &alice_credential_bundle, true)
+        .unwrap();
+
+    let verifiable_group_info = {
+        let serialized = group_info.tls_serialize_detached().unwrap();
+        VerifiableGroupInfo::tls_deserialize(&mut serialized.as_slice()).unwrap()
     };
 
-    group_alice
-        .merge_commit(create_commit_result.staged_commit)
-        .expect("error merging commit");
-
-    let pgs = group_alice
-        .export_public_group_state(backend, &alice_credential_bundle)
-        .expect("Could not export the public group state");
-
-    // Make sure Alice is the signer
-    assert_eq!(pgs.signer, group_alice.own_leaf_index());
-
-    let encoded = pgs.tls_serialize_detached().expect("Could not encode");
-    let verifiable_pgs = VerifiablePublicGroupState::tls_deserialize(&mut encoded.as_slice())
-        .expect("Could not decode");
-
-    let pgs_decoded: PublicGroupState = verifiable_pgs
+    let _: GroupInfo = verifiable_group_info
         .verify(backend, alice_credential_bundle.credential())
-        .expect("error verifiying public group state");
-
-    assert_eq!(pgs, pgs_decoded)
+        .unwrap();
```

Comment:
We have the convention to use `expect` in tests for cases, at least where the checks happen.
This also serves as documentation for what the test is checking.